contrib/pylightning/tests/test_units.py

```diff
@@ -0,0 +1,30 @@
+from lightning import Millisatoshi
+
+
+def test_to_approx_str():
+    amount = Millisatoshi('10000000sat')
+    assert amount.to_approx_str() == "0.1btc"
+    amount = Millisatoshi('1000000sat')
+    assert amount.to_approx_str() == "0.01btc"
+    amount = Millisatoshi('100000sat')
+    assert amount.to_approx_str() == "0.001btc"
+    amount = Millisatoshi('10000sat')
+    assert amount.to_approx_str() == "10000sat"
+    amount = Millisatoshi('1000sat')
+    assert amount.to_approx_str() == "1000sat"
+    amount = Millisatoshi('100msat')
+    assert amount.to_approx_str() == "0.1sat"
+
+    # also test significant rounding
+    amount = Millisatoshi('10001234sat')
+    assert amount.to_approx_str() == "0.1btc"
+    amount = Millisatoshi('1234sat')
+    assert amount.to_approx_str() == "1230sat"
```

Comment:
This is kind of weird, we don't gain a length advantage by rounding the last digit, but we do lose precision.

Reply:
I think the `digit` parameter should allow rounding, but if it doesn't give an advantage length-wise it should force it.

Reply:
fixed